test/runtests.jl

```diff
@@ -1,19 +1,34 @@
 import Enzyme
 import Enzyme_jll
-using ParallelTestRunner: runtests
+using ParallelTestRunner: addworker, filter_tests!, find_tests, parse_args, runtests
 
-function test_filter(test)
-    if test ∈ ("metal", "cuda", "amdgpu")
-        return false
-    end
-    if Sys.iswindows() && test == "ext/specialfunctions"
-        return false
+# Start with autodiscovered tests
+testsuite = find_tests(@__DIR__)
+
+# Parse arguments
+args = parse_args(ARGS)
+
+if filter_tests!(testsuite, args)
+    # Skip GPU-specific tests by default.
+    delete!(testsuite, "metal")
+    delete!(testsuite, "cuda")
+    delete!(testsuite, "amdgpu")
+
+    # Skipped until https://github.com/EnzymeAD/Enzyme.jl/issues/2620 is fixed.
+    if Sys.iswindows()
+        delete!(testsuite, "ext/specialfunctions")
     end
-    if test == "threads"
-        # We run the "threads" tests via the "multi-threads" driver.
-        return false
+
+    # Add threads tests to be run with multiple Julia threads (will be
+    # configured in `test_worker`).
+    testsuite["threads/2"] = :(include($(joinpath(@__DIR__, "threads.jl"))))
```

Comment:
As far as I understand, it's not possible to run just the custom `threads/2` testset from the command line, because when you pass a custom list of testsets as test arguments then `filter_tests!(testsuite, args)` would be false and this line wouldn't be stepped over.  However the user can do
```julia
using Pkg
Pkg.test("Enzyme"; julia_args=`--threads=2`, test_args=`threads`)
```